src/paulscode/android/mupen64plusae/GalleryActivity.java

```diff
@@ -220,6 +271,49 @@ public void onClick( MenuItem menuItem )
             CharSequence message = getText( R.string.invalidInstall_message );
             new Builder( this ).setTitle( title ).setMessage( message ).create().show();
         }
+        
+        if( savedInstanceState != null )
+        {
+            mSelectedItem = null;
+            String md5 = savedInstanceState.getString( STATE_SIDEBAR );
+            if( md5 != null )
+            {
+                // Repopulate the game sidebar
+                for( GalleryItem item : mGalleryItems )
+                {
+                    if( md5.equals( item.md5 ) )
+                    {
+                        onGalleryItemClick( item );
+                        break;
+                    }
+                }
+            }
+            
+            String query = savedInstanceState.getString( STATE_QUERY );
+            if( query != null )
+                mSearchQuery = query;
+        }
+    }
+    
+    @Override
+    public void onSaveInstanceState( Bundle savedInstanceState )
+    {
+        if( mSearchView != null )
+            savedInstanceState.putString( STATE_QUERY, mSearchView.getQuery().toString() );
+        if( mSelectedItem != null )
+            savedInstanceState.putString( STATE_SIDEBAR, mSelectedItem.md5 );
+        
+        super.onSaveInstanceState( savedInstanceState );
+    }
+    
+    public void hideSoftKeyboard()
```

Comment:
Nice bit of polish to close the soft keyboard if it was open (to enter a search query) when opening the sidebar/drawer.